agent/consul/leader_connect_ca_test.go

```diff
@@ -56,8 +56,6 @@ func TestCAManager_Initialize_Vault_Secondary_SharedVault(t *testing.T) {
 		testrpc.WaitForTestAgent(t, serverDC1.RPC, "dc1")
 
 		codec := rpcClient(t, serverDC1)
-		defer codec.Close()
```

Comment:
The `rpcClient` function already does this with `t.Cleanup`

Reply:
TIL about [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup). Thanks! 🙌🏻